org.eclipse.scanning.points/src/org/eclipse/scanning/points/CompoundSpgIteratorFactory.java

```diff
@@ -0,0 +1,173 @@
+/*-
+ *******************************************************************************
+ * Copyright (c) 2011, 2016 Diamond Light Source Ltd.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Matthew Gerring - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.scanning.points;
+
+import static org.eclipse.scanning.points.AbstractScanPointIterator.EMPTY_PY_ARRAY;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.eclipse.scanning.api.points.IDeviceDependentIterable;
+import org.eclipse.scanning.api.points.IMutator;
+import org.eclipse.scanning.api.points.IPointGenerator;
+import org.eclipse.scanning.api.points.IPosition;
+import org.eclipse.scanning.api.points.ScanPointIterator;
+import org.eclipse.scanning.api.points.models.CompoundModel;
+import org.eclipse.scanning.api.points.models.IMapPathModel;
+import org.eclipse.scanning.api.points.models.ScanRegion;
+import org.eclipse.scanning.jython.JythonObjectFactory;
+import org.python.core.PyDictionary;
+import org.python.core.PyList;
+import org.python.core.PyObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ *
+ * @author Matthew Gerring
+ *
+ */
+public class CompoundSpgIteratorFactory {
+
+	protected static class CompoundSgpIterator extends SpgIterator {
+
+		public CompoundSgpIterator(ScanPointIterator pyIterator) {
+			super(pyIterator);
+		}
+
+		@Override
+		public PyDictionary toDict() {
+			return ((PySerializable) pyIterator).toDict();
+		}
+
+	}
+
+	private static Logger logger = LoggerFactory.getLogger(CompoundSpgIteratorFactory.class);
+
+	protected static ScanPointIterator createSpgCompoundGenerator(Iterator<?>[] iterators, Object[] regions,
+			String[] regionAxes, PyObject[] mutators, int duration, boolean continuous) {
+		// TODO Matt D. 2017-10-25: where does this method belong?
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)